zirgen/dsl/src/codegen/_support.rs

```diff
@@ -167,8 +167,14 @@ where
     Ok(output)
 }
 
-pub fn is_nonzero(val: impl Elem) -> bool {
-    val.to_u32_words().into_iter().any(|v| v != 0)
+pub fn is_true(val: impl Elem + From<u32>) -> bool {
+    if val == 0u32.into() {
+        false
+    } else if val == 1u32.into() {
```

Comment:
The `Elem` trait has `ZERO` and `ONE` constants, so I think it would be better to drop the `From<u32>` requirement and use these instead